### PR TITLE
Add section to explain creating classes

### PR DESCRIPTION
**Pull request by garukun**
Added a section to explain the recommended way of creating classes that's best performant in Chrome V8 engines.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -191,6 +192,57 @@ while (keys.length) {
 
 [crockfordconvention]: http://javascript.crockford.com/code.html
 
+## Declare Classes
+
+Declare a class using the `@constructor` jsdoc annotation. Like,
```
Comment (gootik) [inline Readme.md:197]: Do not agree with the usage of annotations.

File: Readme.md [hunk 2]
```diff
@@ -191,6 +192,57 @@ while (keys.length) {
...
+
+```js
+/**
+ * @constructor
+ */
+function SomeClass() {}
+```
+
+While there are several ways to attach methods and properties to an object
+created via `new`, the preferred style for methods is:
```
Comment (gootik) [inline Readme.md:207]: Preferred and correct. If you do it any other way the method will be attached to the instance and not the prototype.

File: Readme.md [hunk 3]
```diff
@@ -191,6 +192,57 @@ while (keys.length) {
...
+
+```js
+SomeClass.prototype.someMethod = function() {
+  /* ... */
+};
+```
+
+The preferred style for other properties is to initialize the field in the
+constructor:
```
Comment (gootik) [inline Readme.md:216]: This is nice. but put a note that the property will be instance local and not prototype local.

File: Readme.md (hunk 1 above)
Comment (garukun) [inline Readme.md:197]: I cannot find the exact article on the performance improvements annotations have on Chrome's V8 engines. But [here](https://developers.google.com/v8/experiments?hl=en) is another article that talks about how folks working on V8 are thinking about introducing another "Strong Mode" to improve the language performance.

In short, I think annotations are are must-have's if not as much as diligently possible. Maybe you can elaborate a bit more on where the disagreement comes from? If it's just styling issue, I think we just need to get pass that. Modern editors like WebStorm does a very good job understanding JsDoc and making code much more readable.

Comment (cndonovan) [inline Readme.md:197]: I am also iffy on this. The downsides I see - more overhead for developer. more weight in the codebase without more needed info. I tend to find Java-style documentation pretty hard to get through - to me, it feels like something that could be said in a single sentence and get the point across fast is belabored and made unnecessarily long.

**Pull request merged**